cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -163,3 +176,82 @@ func (m *AwsManager) GetAsgNodes(asg *Asg) ([]string, error) {
 	}
 	return result, nil
 }
+
+func (m *AwsManager) getAsgTemplate(name string) (*asgTemplate, error) {
+	asg, err := m.service.getAutoscalingGroupByName(name)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceTypeName, err := m.service.getInstanceTypeByLCName(*asg.LaunchConfigurationName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(asg.AvailabilityZones) < 1 {
+		return nil, fmt.Errorf("Unable to get first AvailabilityZone for %s", name)
+	}
+
+	az := *asg.AvailabilityZones[0]
+	region := az[0 : len(az)-1]
+
+	return &asgTemplate{
+		InstanceType: InstanceTypes[instanceTypeName],
+		Region:       region,
+		Tags:         asg.Tags,
+	}, nil
+}
+
+func (m *AwsManager) buildNodeFromTemplate(asg *Asg, template *asgTemplate) (*apiv1.Node, error) {
+	node := apiv1.Node{}
+	nodeName := fmt.Sprintf("%s-asg-%d", asg.Name, rand.Int63())
+
+	node.ObjectMeta = metav1.ObjectMeta{
+		Name:     nodeName,
+		SelfLink: fmt.Sprintf("/api/v1/nodes/%s", nodeName),
+		Labels:   map[string]string{},
+	}
+
+	node.Status = apiv1.NodeStatus{
+		Capacity: apiv1.ResourceList{},
+	}
+
+	// TODO: get a real value.
+	node.Status.Capacity[apiv1.ResourcePods] = *resource.NewQuantity(110, resource.DecimalSI)
+	node.Status.Capacity[apiv1.ResourceCPU] = *resource.NewQuantity(template.InstanceType.VCPU, resource.DecimalSI)
+	node.Status.Capacity[apiv1.ResourceMemory] = *resource.NewQuantity(template.InstanceType.MemoryMb*1024*1024, resource.DecimalSI)
+
+	// TODO: use proper allocatable!!
+	node.Status.Allocatable = node.Status.Capacity
+
+	// NodeLabels
+	node.Labels = cloudprovider.JoinStringMaps(node.Labels, extractLabelsFromAsg(template.Tags))
+	// GenericLabels
+	node.Labels = cloudprovider.JoinStringMaps(node.Labels, buildGenericLabels(template, nodeName))
+
+	node.Status.Conditions = cloudprovider.BuildReadyConditions()
+	return &node, nil
+}
+
+func buildGenericLabels(template *asgTemplate, nodeName string) map[string]string {
+	result := make(map[string]string)
+	// TODO: extract it somehow
+	result[kubeletapis.LabelArch] = cloudprovider.DefaultArch
+	result[kubeletapis.LabelOS] = cloudprovider.DefaultOS
```

Comment:
@mwielgus Would you mind sharing me when we need "correct" values for these labels in practice? Only when the user wants to utilize these in nodeSelector in a pod spec?